runtime-modules/project-token/src/lib.rs

```diff
@@ -0,0 +1,479 @@
+use types::{
+    AccountDataOf, MerkleProofOf, TokenDataOf, TokenIssuanceParametersOf, TransferPolicyOf,
+    TransfersOf,
+};
+
+/// Pallet Configuration Trait
+pub trait Trait: frame_system::Trait {
+    /// Events
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+
+    // TODO: Add frame_support::pallet_prelude::TypeInfo trait
+    /// the Balance type used
+    type Balance: AtLeast32BitUnsigned + FullCodec + Copy + Default + Debug + Saturating + Sum;
+
+    /// The token identifier used
+    type TokenId: AtLeast32BitUnsigned + FullCodec + Copy + Default + Debug;
+
+    /// Block number to balance converter used for interest calculation
+    type BlockNumberToBalance: Convert<Self::BlockNumber, Self::Balance>;
+
+    /// Tresury account for the various tokens
+    type ModuleId: Get<ModuleId>;
+
+    // TODO(after PR round is completed): use Self::ReserveBalance
+    /// Bloat bond value: in JOY
+    type BloatBond: Get<<Self::ReserveCurrency as Currency<Self::AccountId>>::Balance>;
+
+    /// the Currency interface used as a reserve (i.e. JOY)
+    type ReserveCurrency: Currency<Self::AccountId>;
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Token {
+        /// Double map TokenId x AccountId => AccountData for managing account data
+        pub AccountInfoByTokenAndAccount get(fn account_info_by_token_and_account) config():
+        double_map
+            hasher(blake2_128_concat) T::TokenId,
+            hasher(blake2_128_concat) T::AccountId => AccountDataOf<T>;
+
+        /// map TokenId => TokenData to retrieve token information
+        pub TokenInfoById get(fn token_info_by_id) config():
+        map
+            hasher(blake2_128_concat) T::TokenId => TokenDataOf<T>;
+
+        /// Token Id nonce
+        pub NextTokenId get(fn next_token_id) config(): T::TokenId;
+
+        /// Set for the tokens symbols
+        pub SymbolsUsed get (fn symbol_used) config():
+        map
+            hasher(blake2_128_concat) T::Hash => ();
+    }
+
+}
+
+decl_module! {
+    /// _MultiCurrency_ substrate module.
+    pub struct Module<T: Trait> for enum Call
+    where
+        origin: T::Origin
+    {
+
+        /// Default deposit_event() handler
+        fn deposit_event() = default;
+
+        /// Predefined errors.
+        type Error = Error<T>;
+
+        /// Transfer `amount` from `src` account to `dst` according to provided policy
+        /// Preconditions:
+        /// - `token_id` must exists
+        /// - `dst` underlying account must be valid for `token_id`
+        /// - `src` must be valid for `token_id`
+        /// - `dst` is compatible con `token_id` transfer policy
+        ///
+        /// Postconditions:
+        /// - `src` free balance decreased by `amount` or removed if final balance < existential deposit
+        /// - `dst` free balance increased by `amount`
+        /// - `token_id` issuance eventually decreased by dust amount in case of src removalp
```

Comment:
:+1:

Reply:
Partially fixed, there are still mentions of existential deposit in postconditions:
- ```free balance decreased by `amount` or removed if final balance < existential deposit```
- ``` `token_id` issuance eventually decreased by dust amount in case of src removalp```